core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -546,7 +546,7 @@ private[spark] class ExecutorMonitor(
           } else {
             Long.MaxValue
           }
-          math.min(_cacheTimeout, _shuffleTimeout)
+          math.max(_cacheTimeout, _shuffleTimeout)
```

Comment:
When `cachedBlocks.empty == true`, this will be `Long.MaxValue`. That's wrong, isn't it, @hanyucui ?

Reply:
@dongjoon-hyun Yeah, you are right. This is not a solution but only an illustration. I wish there was a way to prevent this from bothering admins.

In my case, I enable dynamic allocation on K8s. I set `spark.dynamicAllocation.shuffleTracking.timeout` to a small number so that I can kill the executors earlier but set `spark.dynamicAllocation.cachedExecutorIdleTimeout` to a larger value since some users would like to keep their cached data around for longer. I expected the executors to be killed when the larger timeout is hit, not the smaller one, but the current behavior is the opposite. I understand different users have different requirements and this might totally make sense for them. No action is needed on your side -- I will think it through first.

Reply:
I understand your opinion, but this feature is not designed like that originally.